lib/config/options/index.ts

```diff
@@ -2844,6 +2844,13 @@ const options: RenovateOptions[] = [
     cli: false,
     env: false,
   },
+  {
+    name: 'milestone',
+    description: `The number of a milestone. If set, the milestone will be set on the PR.`,
```

Comment:
```suggestion
    description: `The number of a milestone. If set, the milestone will be set when Renovate creates the PR.`,
```

We should probably mention the "Renovate only adds the milestone when creating the PR" thing in the description as well.